drivers/espi/espi_mchp_xec.c

```diff
@@ -0,0 +1,831 @@
+						ESPI_SLAVE_TO_MASTER },
+	[ESPI_VWIRE_SIGNAL_SLV_BOOT_STS]  = { SMVW01, ESPI_VWIRE_SRC_ID3,
+						ESPI_SLAVE_TO_MASTER },
+	/* SMVW02 */
+	[ESPI_VWIRE_SIGNAL_SCI]           = { SMVW02, ESPI_VWIRE_SRC_ID0,
+						ESPI_SLAVE_TO_MASTER },
+	[ESPI_VWIRE_SIGNAL_SMI]           = { SMVW02, ESPI_VWIRE_SRC_ID1,
+						ESPI_SLAVE_TO_MASTER },
+	[ESPI_VWIRE_SIGNAL_RST_CPU_INIT]  = { SMVW02, ESPI_VWIRE_SRC_ID2,
+						ESPI_SLAVE_TO_MASTER },
+	[ESPI_VWIRE_SIGNAL_HOST_RST_ACK]  = { SMVW02, ESPI_VWIRE_SRC_ID3,
+						ESPI_SLAVE_TO_MASTER },
+	/* MSVW02 */
+	[ESPI_VWIRE_SIGNAL_HOST_RST_WARN] = { MSVW02, ESPI_VWIRE_SRC_ID0,
+						ESPI_MASTER_TO_SLAVE },
+	/* SMVW03 */
+	[ESPI_VWIRE_SIGNAL_SUS_ACK]       = { SMVW03, ESPI_VWIRE_SRC_ID0,
+						ESPI_SLAVE_TO_MASTER },
+	[ESPI_VWIRE_SIGNAL_DNX_ACK]       = { SMVW03, ESPI_VWIRE_SRC_ID1,
+						ESPI_SLAVE_TO_MASTER },
+	/* MSVW03 */
+	[ESPI_VWIRE_SIGNAL_SUS_WARN]      = { MSVW03, ESPI_VWIRE_SRC_ID0,
+						ESPI_MASTER_TO_SLAVE },
+	[ESPI_VWIRE_SIGNAL_SUS_PWRDN_ACK] = { MSVW03, ESPI_VWIRE_SRC_ID1,
+						ESPI_MASTER_TO_SLAVE },
+	[ESPI_VWIRE_SIGNAL_SLP_A]         = { MSVW03, ESPI_VWIRE_SRC_ID3,
+						ESPI_MASTER_TO_SLAVE },
+	/* MSVW04 */
+	[ESPI_VWIRE_SIGNAL_SLP_LAN]       = { MSVW04, ESPI_VWIRE_SRC_ID0,
+						ESPI_MASTER_TO_SLAVE },
+	[ESPI_VWIRE_SIGNAL_SLP_WLAN]      = { MSVW04, ESPI_VWIRE_SRC_ID1,
+						ESPI_MASTER_TO_SLAVE },
+	/* MSVW07 */
+	[ESPI_VWIRE_SIGNAL_HOST_C10]      = { MSVW07, ESPI_VWIRE_SRC_ID0,
+						ESPI_MASTER_TO_SLAVE },
+	/* MSVW08 */
+	[ESPI_VWIRE_SIGNAL_DNX_WARN]      = { MSVW08, ESPI_VWIRE_SRC_ID1,
+						ESPI_MASTER_TO_SLAVE },
+};
+
+static int espi_xec_configure(struct device *dev, struct espi_cfg *cfg)
+{
+	u8_t iomode = 0;
+	u8_t cap1 = ESPI_CAP_REGS->GLB_CAP1;
+	u8_t cur_iomode = (cap1 & MCHP_ESPI_GBL_CAP1_IO_MODE_MASK) >>
+		MCHP_ESPI_GBL_CAP1_IO_MODE_POS;
+
+	/* Set frequency */
+	cap1 &= ~MCHP_ESPI_GBL_CAP1_MAX_FREQ_MASK;
+	switch (cfg->max_freq) {
+	case 20:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_20M;
+		break;
+	case 25:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_25M;
+		break;
+	case 33:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_33M;
+		break;
+	case 50:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_50M;
+		break;
+	case 66:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_66M;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	/* Set IO mode */
+	iomode = (cfg->io_caps >> 1);
+	if (iomode > 3) {
+		return -EINVAL;
+	}
+
+	if (iomode != cur_iomode) {
+		cap1 &= ~MCHP_ESPI_GBL_CAP1_IO_MODE_MASK0 <<
+			MCHP_ESPI_GBL_CAP1_IO_MODE_POS;
+		cap1 |= (iomode << MCHP_ESPI_GBL_CAP1_IO_MODE_POS);
+	}
+
+	ESPI_CAP_REGS->GLB_CAP1 = cap1;
+
+	/* Activate the eSPI block *.
+	 * Need to guarantee that this register is configured before RSMRST#
+	 * de-assertion and after pinmux
+	 */
+	ESPI_EIO_BAR_REGS->IO_ACTV = 1;
+	LOG_DBG("eSPI block activated successfully\n");
+	return 0;
+}
+
+static bool espi_xec_channel_ready(struct device *dev,
+					enum espi_channel ch)
```

Comment:
there is still indentation issue: that parameter should be below the 's' of struct. (so below last '(' + 1 space).ù

Your editor should be doing that for you if you configure it to follow linux code style.

Reply:
Are you suggesting to mix tabs and spaces here? I haven't seen that anywhere in the rest of Zephyr code.

Reply:
Addressed anyways to follow linux kernel closely

Reply:
tabs and spaces afterwards are hopefully allowed and everywhere in zephyr.

spaces + tabs or tabs + spaces + tabs are not.